cycode/cli/printers/base_table_printer.py

```diff
@@ -0,0 +1,112 @@
+import abc
+from typing import List, NamedTuple, Dict, Optional
+
+import click
+from texttable import Texttable
+
+from cycode.cli.consts import SECRET_SCAN_TYPE, INFRA_CONFIGURATION_SCAN_TYPE, SAST_SCAN_TYPE
+from cycode.cli.printers.text_printer import TextPrinter
+from cycode.cli.models import DocumentDetections, CliError, CliResult
+from cycode.cli.printers.base_printer import BasePrinter
+
+
+class ColumnInfo(NamedTuple):
+    name: str
+    index: int  # lower index means left column
+    width_secret: int = 1
+    width_sast: int = 1
+    width_iac: int = 1
```

Comment:
I think that its better to remove the mention of the scan types
the tuple should be (name,index,width)
calculate the width by the scan type before creating the ColumnInfo
you can remove also the _get_table_columns_width func that has mentions to scan types as well